src/evo/deterministicmns.cpp

```diff
@@ -247,7 +250,7 @@ std::vector<CDeterministicMNCPtr> CDeterministicMNList::GetProjectedMNPayees(int
         return CompareByLastPaid(a.get(), b.get());
     });
 
-    result.resize(nCount);
+    result.resize(nCount - (GetMnType(MnType::HighPerformance).voting_weight - remaining_hpmn_payments));
```

Comment:
why?

Reply:
Imagine a network with 2 MNs (a, b) and 2 HPMNs (C, D).
`GetValidWeightedMNsCount` returns 10.

Suppose that tip height is 100 and that the last payee is C with 2 remaining payments.

```
99: C
100: C (tip)
101: C (1st extra payment)
102: C (2nd extra payment)
(At this point we skip C)
103: a
104: D
105: D
106: D
107: D
108: b
```
Note that 6 new payments were added + the 2 remaining from C = 8 (while `GetValidWeightedMNsCount` returns 10)

If we discard the change: 
`result.resize(nCount - (GetMnType(MnType::HighPerformance).voting_weight - remaining_hpmn_payments));` then we will also get in return:

```
109: UNKNOWN
110: UNKNOW
```